app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -0,0 +1,67 @@
+package net.squanchy.schedule
+
+import io.reactivex.Observable
+import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.schedule.domain.view.Schedule
+import net.squanchy.schedule.domain.view.SchedulePage
+import net.squanchy.service.firebase.FirebaseAuthService
+import net.squanchy.service.firebase.FirebaseDbService
+import net.squanchy.service.firebase.model.FirebaseDays
+import net.squanchy.service.repository.EventRepository
+import org.joda.time.LocalDateTime
+import org.joda.time.format.DateTimeFormat
+
+class ScheduleService internal constructor(
+        private val dbService: FirebaseDbService,
+        private val authService: FirebaseAuthService,
+        private val eventRepository: EventRepository
+) {
+
+    fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
+        return authService.ifUserSignedInThenObservableFrom { userId ->
+            val daysObservable = dbService.days()
+
+            val eventsObservable = eventRepository.events(userId)
+                    .map { events -> if (onlyFavorites) events.filter { it.favorited } else events }
+                    .map { it.groupBy { it.dayId } }
+
+            eventsObservable
+                    .withLatestFrom(daysObservable, combineEventsById())
+                    .subscribeOn(Schedulers.io())
+        }
+    }
+
+    private fun combineEventsById(): BiFunction<Map<String, List<Event>>, FirebaseDays, Schedule> {
+        return BiFunction { eventsMap, (days) ->
+            val pages = eventsMap.keys
+                    .map { key ->
```

Comment:
Can you split this into a separate function? It looks a bit hard to understand to me